cmd/tether/main_darwin.go

```diff
@@ -69,7 +66,11 @@ func main() {
 		return
 	}
 
-	err = run(src, sink)
+	// create the tether and register the attach extension
+	tthr := tether.New(src, sink, &operations{})
+	tthr.Register("Attach", sshserver)
```

Comment:
I'm not expecting any currently but I've also not thought about what else could be added. This is here because:
a. it provides a mechanism for dependency injection during testing
b. it means we can use the core of the tether code without needing to implement backchannel and corresponding supporting functions when it's not needed (#746)